Crash when extending a type alias to a trait with implicit parameter

I found this issue while trying to write some tests for issue #4582. PR #4827 is related.
```scala
trait T[X]
type Foo[X, Y] = T[X]
class D[X] extends Foo[X, Unit]
```
```scala
trait T[X: Numeric]
type Foo[X] = T[X]
class D[X: Numeric] extends Foo[X]
```
Two above examples were compiled.
```scala
trait T[X: Numeric]
type Foo[X, Y] = T[X]
class D[X: Numeric] extends Foo[X, Unit]
```
However, this one resulted the following crash:
```scala
Exception in thread "main" java.lang.AssertionError: assertion failed: missing parameters for trait T from ($outer: C, implicit evidence$2: scala.math.Numeric) extends Object() with T {
  private implicit val evidence$2: scala.math.Numeric
  private val $outer: C
  private <accessor> def $outer(): C = this.$outer
  final def C$D$$$outer(): C = D.this.$outer()
} should have been caught in typer
	at scala.Predef$.assert(Predef.scala:219)
	at dotty.tools.dotc.transform.Mixin.nextArgument$1(Mixin.scala:209)
	at dotty.tools.dotc.transform.Mixin.$anonfun$transformTemplate$12(Mixin.scala:235)
	at scala.collection.TraversableLike$WithFilter.$anonfun$map$2(TraversableLike.scala:739)
	at scala.collection.immutable.List.foreach(List.scala:389)
	at scala.collection.TraversableLike$WithFilter.map(TraversableLike.scala:738)
	at dotty.tools.dotc.transform.Mixin.traitInits$1(Mixin.scala:217)
	at dotty.tools.dotc.transform.Mixin.$anonfun$transformTemplate$17(Mixin.scala:263)
	at scala.collection.immutable.List.flatMap(List.scala:335)
	at dotty.tools.dotc.transform.Mixin.transformTemplate(Mixin.scala:262)
	at dotty.tools.dotc.transform.Mixin.transformTemplate(Mixin.scala:98)
	at dotty.tools.dotc.transform.MegaPhase.goTemplate(MegaPhase.scala:922)
```

--- a/compiler/src/dotty/tools/dotc/typer/Typer.scala
+++ b/compiler/src/dotty/tools/dotc/typer/Typer.scala
@@ -1501,7 +1501,7 @@ class Typer extends Namer
      */
     def maybeCall(ref: Tree, psym: Symbol, cinfo: Type): Tree = cinfo.stripPoly match {
       case cinfo @ MethodType(Nil) if cinfo.resultType.isImplicitMethod =>
-        typedExpr(untpd.New(ref, Nil))(superCtx)
+        typedExpr(untpd.New(untpd.TypedSplice(ref)(superCtx), Nil))(superCtx)
       case cinfo @ MethodType(Nil) if !cinfo.resultType.isInstanceOf[MethodType] =>
         ref
       case cinfo: MethodType =>
@@ -1518,7 +1518,7 @@ class Typer extends Namer
 
     def typedParent(tree: untpd.Tree): Tree = {
       var result = if (tree.isType) typedType(tree)(superCtx) else typedExpr(tree)(superCtx)
-      val psym = result.tpe.typeSymbol
+      val psym = result.tpe.dealias.typeSymbol
       if (seenParents.contains(psym) && !cls.isRefinementClass)
         ctx.error(i"$psym is extended twice", tree.pos)
       seenParents += psym

--- a/tests/neg/i4837.scala
+++ b/tests/neg/i4837.scala
@@ -1,0 +1,6 @@
+trait T(x: Int)
+
+class C {
+  type Foo[X] = T
+  class D extends Foo[Unit] // error
+}

--- a/tests/pos/i4837.scala
+++ b/tests/pos/i4837.scala
@@ -1,0 +1,11 @@
+trait T[X: Numeric]
+
+class C {
+  type S[X] = T[X]
+  type Foo[X, Y] = T[X]
+  type Bar[X, Y] = Foo[X, Y]
+
+  class D[X: Numeric] extends S[X]
+  class E[X: Numeric] extends Foo[X, Unit]
+  class F[X: Numeric] extends Bar[X, Unit]
+}

